kotlinx-coroutines-core/common/src/CoroutineStart.kt

```diff
@@ -48,10 +48,29 @@ public enum class CoroutineStart {
      * Atomically (i.e., in a non-cancellable way) schedules the coroutine for execution according to its context.
      * This is similar to [DEFAULT], but the coroutine cannot be cancelled before it starts executing.
      *
+     * The coroutine stated with [ATOMIC] is guaranteed to start execution even if its [Job] was cancelled.
+     * This [CoroutineStart] option can be used to ensure resources' disposal in case of cancellation.
+     * For example, closing a channel used by a producer:
```

Comment:
Is this a complete sentence? Maybe explain the idea more verbosely, like "For example, this `producer` guarantees that the channel will be eventually closed, even if the coroutine scope is cancelled before `producer` is called:"?